benchmarks/src/lib.rs

```diff
@@ -15,4 +15,116 @@
 // specific language governing permissions and limitations
 // under the License.
 
+use datafusion::DATAFUSION_VERSION;
+use serde::Serialize;
+use serde_json::Value;
+use std::{collections::HashMap, time::SystemTime};
+
 pub mod tpch;
+
+#[derive(Debug, Serialize)]
+pub struct RunContext {
+    /// Benchmark crate version
+    pub benchmark_version: String,
+    /// DataFusion crate version
+    pub datafusion_version: String,
+    /// Number of CPU cores
+    pub num_cpus: usize,
+    /// Start time
+    pub start_time: u64,
+    /// CLI arguments
+    pub arguments: Vec<String>,
+}
+
+impl Default for RunContext {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl RunContext {
+    pub fn new() -> Self {
+        Self {
+            benchmark_version: env!("CARGO_PKG_VERSION").to_owned(),
+            datafusion_version: DATAFUSION_VERSION.to_owned(),
+            num_cpus: num_cpus::get(),
+            start_time: SystemTime::now()
+                .duration_since(SystemTime::UNIX_EPOCH)
+                .expect("current time is later than UNIX_EPOCH")
+                .as_secs(),
+            arguments: std::env::args().skip(1).collect::<Vec<String>>(),
+        }
+    }
+}
+
+/// A single iteration of a benchmark query
+#[derive(Debug, Serialize)]
+struct QueryIter {
+    elapsed: f64,
+    row_count: usize,
+}
+/// A single benchmark case
+#[derive(Debug, Serialize)]
+pub struct BenchQuery {
+    query: String,
+    iterations: Vec<QueryIter>,
+    start_time: u64,
```

Comment:
I see this was just carried over from the previous approach, but I wonder why it needs to be converted to `u64` (and loses the fact that this is now epoch seconds)

Perhaps we can either keep the field as `SystemTime` or document what unit (seconds since epoch) this is?

Reply:
sure that sounds good to me